Sources/MongoSwift/BSON/Document+Collection.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+/// An extension of `Document` to make it conform to the `Collection` protocol.
+/// This gives guarantees on non-destructive iteration, and offers an indexed
+/// ordering to the key value pairs in the document.
+extension Document: Collection {
+    /// Returns the start index of the Document.
+    public var startIndex: Int {
+        precondition(self.count > 0)
```

Comment:
Hm, this is probably a small issue in their documentation. I actually distinctly recall reading "The position of the first element in a nonempty collection",  and going "okay, that sounds like a precnodition!". 😆 

It seems like this should actually just return `endIndex` instead... hard to really tell since its contradictory.

Since I don't like `precondition` or `preconditionFailure` and prefer to avoid them whenever possible, I'm going to switch this over to returning `endIndex`. The cool thing here is that due to how I've chosen indices, we can just have `startIndex` always return 0.